src/wallet/spend.cpp

```diff
@@ -566,15 +566,18 @@ std::optional<SelectionResult> SelectCoins(const CWallet& wallet, const std::vec
             }
         }
         // Coin Selection failed.
-        return std::optional<SelectionResult>();
+        return CallResult<SelectionResult>();
     }();
 
-    if (!res) return std::nullopt;
+    if (!res) {
+        return res;
+    }
 
     // Add preset inputs to result
-    res->AddInput(preset_inputs);
+    auto resMerged = *res.getObjResult();
```

Comment:
In c2ec6661cbb70a388dc67d6b4b47f2ab49b99418 "wallet: make SelectCoins return a CallResult."

nit: We prefer snake_case for new variables. (here and elsewhere)